x-pack/plugins/security_solution/server/lib/detection_engine/signals/bulk_create_threshold_signals.ts

```diff
@@ -116,10 +141,16 @@ const getTransformedHits = (
       return [];
     }
 
+    const hit = results.hits.hits[0];
+    if (hit == null) {
+      logger.warn(`No hits returned, but totalResults >= threshold.value (${threshold.value})`);
+      return [];
+    }
+
     const source = {
-      '@timestamp': new Date().toISOString(),
+      '@timestamp': get(timestampOverride ?? '@timestamp', hit._source),
```

Comment:
Does this set the threshold signal date time to be the one from the hit?

I ask because previously there has been some debate previously about what a signal's `@timestamp` should be as some people prefer the source signal timestamp instead of the current date time that the signal is created but the decision so far has been that the `@timestamp` for a signal should be the current date time stamp and then if a rule has an `original_timestamp` it would be stored underneath `signal.original_time`

Threshold is unique in that is synthetic but the `@timestamp` should still be the timestamp of when the signal is created for consistency. That doesn't mean that we can't change them all as part of 

<img width="2256" alt="Screen Shot 2020-11-10 at 7 04 55 PM" src="https://user-images.githubusercontent.com/1151048/98756464-db325c80-2387-11eb-9906-663867ad6176.png">

Reply:
@FrankHassanabad I thought that too initially, but the signal timestamp is created later. This timestamp is later used to populated `signal.original_time` (it will be the original time of the last event in the timespan... https://github.com/elastic/kibana/blob/b6d661f9c38f1c83a747e98b99594ecc8b567b76/x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_signal.ts#L99) and the signal timestamp is created here (https://github.com/elastic/kibana/blob/b6d661f9c38f1c83a747e98b99594ecc8b567b76/x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_bulk_body.ts#L77).

Reply:
Oh I see, thanks. Yeah, that makes sense. I get it.